.github/workflows/reusable_release_automation.yaml

```diff
@@ -0,0 +1,115 @@
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+    secrets:
+      bot_token:
+        description: "github token"
+        required: true
+      slack_channel:
+        description: "slack channel for sending a message in case of failure"
+        required: true
+      slack_token:
+        description: "slack token for sending the message"
+        required: true
+
+jobs:
+  check_block_endpoint:
+    name: Check block endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    needs: check_block_endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.bot_token }}"
+        run: |
+          # Get the date of the latest release.
+          LATEST_RELEASE=$(
+            gh release list --json publishedAt,isLatest -R "${{ github.repository }}" --jq '
+              .[] | select(.isLatest) | .publishedAt | fromdateiso8601
+            '
+          )
+
+          # List all releases, filter by prereleases that was made later than the latest release, sort by date,
+          # reverse because the last one is the most recent and only grab the last prerelease
+          NON_PUBLISHED_PRERELEASE=$(
+            gh release list --json name,publishedAt,isPrerelease,isLatest,tagName -R "${{ github.repository }}" --jq '
+              [
+                  .[] | select( .isPrerelease and ((.publishedAt | fromdateiso8601) > '${LATEST_RELEASE}'))
+              ]
+              | sort_by(.publishedAt |= fromdateiso8601) | reverse | first
+            '
+          )
+
+          # Get prerelease name
+          NON_PUBLISHED_PRERELEASE_NAME=$(echo "$NON_PUBLISHED_PRERELEASE" | jq -r '.name')
+
+          # Check specifically for "artifacts-pending" or "pre-release-failure" conditions
+          if [[ "$NON_PUBLISHED_PRERELEASE_NAME" == *"artifacts-pending"* || "$NON_PUBLISHED_PRERELEASE_NAME" == *"pre-release-failure"* ]]; then
+            echo "Prerelease $NON_PUBLISHED_PRERELEASE_NAME is blocked due to 'artifacts-pending' or 'pre-release-failure'."
+            # Setting an environment variable for conditional steps later on
+            echo "SHOULD_NOTIFY=1" >> $GITHUB_ENV
+          else
+            PRERELEASE_TAG=$(echo "$NON_PUBLISHED_PRERELEASE" | jq -r '.tagName')
+
+            # Promote the prerelease to a full release
+            gh release edit "$PRERELEASE_TAG" --repo "${{ github.repository }}" --draft=false --prerelease=false
+
+            echo "SHOULD_NOTIFY=0" >> $GITHUB_ENV
+          fi
```

Comment:
Did you consider releasing only if the name matches (instead of check for specific conditions)? I mean something like this:

```bash
# NON_PUBLISHED_TAG_NAME should be in the API response
if [[ $NON_PUBLISHED_PRERELEASE_NAME == $NON_PUBLISHED_TAG_NAME ]]; then
  # should be promoted
else
  # should notify
fi
```

That way, blocking a pre-release manually would be less error-prone. Eg: we manually add `(artifacts_pending)` instead of `(artifacts-pending)` to the release title. We could also use other conventions such us `(testing)` to _block_ manually a specific pre-release.

Reply:
Less error-prone 👍 Done in 4b2e6e0f22427753e337821f4ef0b190c1db4fde.